src/lineup/internal/panel/SearchBoxProvider.ts

```diff
@@ -0,0 +1,48 @@
+import {SearchBox, LocalDataProvider, IGroupSearchItem, ISearchBoxOptions} from 'lineupjs';
+import {ISearchOption} from '../LineUpPanelActions';
+import i18n from 'phovea_core/src/i18n';
+
+/**
+ * The SearchBoxProvider allows creating multiple LineUp SearchBoxes and stores them internally in a list.
+ * All created search boxes can be updated simultaneously with a list of searchable items.
+ */
+export default class SearchBoxProvider {
+
+  /**
+   * List of created LineUp SearchBoxes
+   */
+  private searchBoxes: SearchBox<ISearchOption>[] = [];
+
+  get length(): number {
+    return this.searchBoxes.length;
+  }
+
+  /**
+   * Create a new LineUp SearchBox. The instance is added to the internal list and returned.
+   * @returns A new LineUp SearchBox instance
+   */
+  createSearchBox(options?: Partial<ISearchBoxOptions<ISearchOption>>): SearchBox<ISearchOption> {
```

Comment:
I just tested what happens when you pass `undefined` as a parameter to `Object.assign()`. it works, however I think it would be better to have the `options` parameter initialized as an empty object

Reply:
Good point! I changed it.